packages/salesforce-adapter/src/config_change.ts

```diff
@@ -86,24 +89,104 @@ export const createRetrieveConfigChange = (result: RetrieveResult): ConfigChange
       regexRes?.groups?.instance as string
     ))
 
+const convertDeprecatedRegex = (filePathRegex: string): string => {
+  let newPathRegex = filePathRegex
+
+  newPathRegex = filePathRegex.startsWith('^')
+    ? newPathRegex.substring(1)
+    : newPathRegex
+
+  newPathRegex = !filePathRegex.startsWith('.*') && !filePathRegex.startsWith('^')
+    ? `.*${newPathRegex}`
+    : newPathRegex
+
+  newPathRegex = filePathRegex.endsWith('$')
+    ? newPathRegex.substring(0, newPathRegex.length - 1)
+    : newPathRegex
+
+  newPathRegex = !filePathRegex.endsWith('$') && (!filePathRegex.endsWith('.*') || filePathRegex.endsWith('\\.*'))
+    ? `${newPathRegex}.*`
+    : newPathRegex
+
+  return newPathRegex
+}
+
+export const convertDeprecatedDataConf = (conf: DataManagementConfig): DataManagementConfig => {
+  const updatedConf = _.cloneDeep(conf)
+  updatedConf.includeObjects = updatedConf.includeObjects.map(convertDeprecatedRegex)
+  updatedConf.excludeObjects = updatedConf?.excludeObjects?.map(convertDeprecatedRegex)
+  updatedConf.allowReferenceTo = updatedConf?.allowReferenceTo?.map(convertDeprecatedRegex)
+  updatedConf.saltoIDSettings.overrides = updatedConf.saltoIDSettings.overrides?.map(
+    override => ({ ...override, objectsRegex: convertDeprecatedRegex(override.objectsRegex) })
+  )
+  return updatedConf
+}
```

Comment:
less side effects and mutation in-place, avoid `cloneDeep` whenever possible
```suggestion
export const convertDeprecatedDataConf = (conf: DataManagementConfig): DataManagementConfig => ({
  ...conf,
  includeObjects: conf.includeObjects.map(convertDeprecatedRegex),
  excludeObjects: conf?.excludeObjects?.map(convertDeprecatedRegex),
  allowReferenceTo: conf?.allowReferenceTo?.map(convertDeprecatedRegex)
  saltoIDSettings: {
    ...conf.saltoIDSettings,
    overrides: conf.saltoIDSettings.overrides?.map(
      override => ({ ...override, objectsRegex: convertDeprecatedRegex(override.objectsRegex) })
    ),
  },
})
```